pymeasure/instruments/newport/smc100.py

```diff
@@ -0,0 +1,706 @@
+        "KP?", "KP%g",
+        """Get or set the proportional gain of the PID control loop.
+
+        Type: Floating point number
+        Range: [0, 1e12[
+        Unit: V/u""",
+        validator=strict_range,
+        values=(0, 1e12),
+        check_get_errors=True,
+        check_set_errors=True
+    )
+    position = Instrument.control(
+        "TP", "PA%g",
+        """Get current position or move to set position.
+
+        Type: Floating point number
+        Range: [left software limit, right software limit]
+        Unit: u
+
+        Example use:
+
+        .. code-block:: python
+
+            smc.position
+            smc.position = 10
+            smc.position += 1""",
+        check_get_errors=True,
+        check_set_errors=True
+    )
+    position_theoretical = Instrument.measurement(
+        "TH",
+        """Get the theoretical position.
+
+        Type: Floating point number
+        Range: [left software limit, right software limit]
+        Unit: u""",
+        check_get_errors=True
+    )
+    revision = Instrument.measurement(
+        "VE",
+        """Get the revision information of the controller.""",
+        check_get_errors=True,
+        cast=str
+    )
+    state = Instrument.measurement(
+        "TS",
+        """Get the controller state.
+
+        Be careful: this command erases the last positioner error, which
+        should be checked first if it is of interest.""",
+        get_process=lambda v: ControllerState(v[-2:]),
+        check_get_errors=True,
+        cast=str
+    )
+    unit = Instrument.control(
+        "SU?", "SU%g",
+        """Get or set the value for one encoder count in the unit which
+        is thus defined.
+
+        Type: Floating point number
+        Range: ]1e-6, 1e12[
+        Unit: u""",
+        validator=strict_range,
+        values=(1e-6, 1e12),
+        check_get_errors=True,
+        check_set_errors=True
+    )
+    velocity = Instrument.control(
+        "VA?", "VA%g",
+        """Get or set the velocity.
+
+        Type: Floating point number
+        Range: ]1e-6, 1e12[
+        Unit: u/s""",
+        validator=strict_range,
+        values=(1e-6, 1e12),
+        check_get_errors=True,
+        check_set_errors=True
+    )
+    voltage = Instrument.control(
+        "DV?", "DV%g",
+        """Get or set the maximum driver voltage.
+
+        Type: Floating point number
+        Range: [12, 48]
+        Unit: V""",
+        validator=strict_range,
+        values=(12, 24),
+        check_get_errors=True,
+        check_set_errors=True
+    )
+
+    # Special methods
+    def __init__(self, adapter, address=1, name="SMC100", **kwargs):
+        """Set up connection to the controller."""
+        self.address = str(address)
+        super().__init__(
+            adapter,
+            name,
+            includeSCPI=False,
+            baud_rate=57600,
+            flow_control=ControlFlow.xon_xoff,
+            preprocess_reply=lambda v, address=self.address: sub(
```

Comment:
Ha, two actually -- we are deprecating a global `preprocess_reply`, this should go in SMC100.read, instead.